pkg/kapp/cmd/app/apply_flags.go

```diff
@@ -55,6 +55,8 @@ func (s *ApplyFlags) SetWithDefaults(prefix string, defaults ApplyFlags, cmd *co
 
 	cmd.Flags().DurationVar(&s.WaitingChangesOpts.Timeout, prefix+"wait-timeout",
 		mustParseDuration("15m"), "Maximum amount of time to wait in wait phase")
+	cmd.Flags().DurationVar(&s.WaitingChangesOpts.ResourceTimeout, prefix+"wait-resource-timeout",
+		mustParseDuration("15m"), "Maximum amount of time to wait for a resource in wait phase")
```

Comment:
i was thinking a bit more about this. i think we should set this to default to 0s (also add a note to flag description, 0s means no timeout). check in WaitingChanges if duration is 0 and skip timing out if it is. reasoning behind this is that we currently dont have per resource timeout, and i dont think we want to add a timeout since it may affect some existing deployments.

Reply:
Yeah sure sounds reasonable. Done